File: BC-W4/Army/Abilities/BerserkerAbility/BerserkerAbility.cpp
#include "BerserkerAbility.h"

BerserkerAbility::BerserkerAbility(Unit* unit) : Ability(unit) {};
BerserkerAbility::~BerserkerAbility() {};

void BerserkerAbility::castAbility() {
    throw HasNoSuchAbilityException();
};

void BerserkerAbility::castAbility(Unit* enemy) {
    throw HasNoSuchAbilityException();
};

File: BC-W4/ArmyJava/src/abilities/BerserkerAbility.java
package abilities;

import units.Unit;
import exceptions.*;

public class BerserkerAbility extends Ability {
    public BerserkerAbility(Unit owner) {
        super(owner);
    }

    public void transform() throws InvalidTransformException {
        throw new InvalidTransformException();
    }

    public void infect(Unit enemy) throws InvalidInfectException {
        throw new InvalidInfectException();
    }
}
